meson.build

```diff
@@ -13,256 +104,477 @@ uvsrc = [
+    'test/test-shutdown-eof.c',
+    'test/test-shutdown-twice.c',
+    'test/test-signal-multiple-loops.c',
+    'test/test-signal-pending-on-close.c',
+    'test/test-signal.c',
+    'test/test-socket-buffer-size.c',
+    'test/test-spawn.c',
+    'test/test-stdio-over-pipes.c',
+    'test/test-strscpy.c',
+    'test/test-tcp-alloc-cb-fail.c',
+    'test/test-tcp-bind-error.c',
+    'test/test-tcp-bind6-error.c',
+    'test/test-tcp-close-accept.c',
+    'test/test-tcp-close-while-connecting.c',
+    'test/test-tcp-close.c',
+    'test/test-tcp-close-reset.c',
+    'test/test-tcp-connect-error-after-write.c',
+    'test/test-tcp-connect-error.c',
+    'test/test-tcp-connect-timeout.c',
+    'test/test-tcp-connect6-error.c',
+    'test/test-tcp-create-socket-early.c',
+    'test/test-tcp-flags.c',
+    'test/test-tcp-oob.c',
+    'test/test-tcp-open.c',
+    'test/test-tcp-read-stop.c',
+    'test/test-tcp-read-stop-start.c',
+    'test/test-tcp-shutdown-after-write.c',
+    'test/test-tcp-try-write.c',
+    'test/test-tcp-try-write-error.c',
+    'test/test-tcp-unexpected-read.c',
+    'test/test-tcp-write-after-connect.c',
+    'test/test-tcp-write-fail.c',
+    'test/test-tcp-write-queue-order.c',
+    'test/test-tcp-write-to-half-open-connection.c',
+    'test/test-tcp-writealot.c',
+    'test/test-test-macros.c',
+    'test/test-thread-equal.c',
+    'test/test-thread.c',
+    'test/test-threadpool-cancel.c',
+    'test/test-threadpool.c',
+    'test/test-timer-again.c',
+    'test/test-timer-from-check.c',
+    'test/test-timer.c',
+    'test/test-tmpdir.c',
+    'test/test-tty-duplicate-key.c',
+    'test/test-tty-escape-sequence-processing.c',
+    'test/test-tty.c',
+    'test/test-udp-alloc-cb-fail.c',
+    'test/test-udp-bind.c',
+    'test/test-udp-connect.c',
+    'test/test-udp-create-socket-early.c',
+    'test/test-udp-dgram-too-big.c',
+    'test/test-udp-ipv6.c',
+    'test/test-udp-mmsg.c',
+    'test/test-udp-multicast-interface.c',
+    'test/test-udp-multicast-interface6.c',
+    'test/test-udp-multicast-join.c',
+    'test/test-udp-multicast-join6.c',
+    'test/test-udp-multicast-ttl.c',
+    'test/test-udp-open.c',
+    'test/test-udp-options.c',
+    'test/test-udp-send-and-recv.c',
+    'test/test-udp-send-hang-loop.c',
+    'test/test-udp-send-immediate.c',
+    'test/test-udp-sendmmsg-error.c',
+    'test/test-udp-send-unreachable.c',
+    'test/test-udp-try-send.c',
+    'test/test-uname.c',
+    'test/test-walk-handles.c',
+    'test/test-watcher-cross-stop.c',
+  ]
+  if win32
+    # TODO: not sure if this works like that, it doesn't on linux
+    libuv_test_src += meson.source_root() + '/uv_win_longpath.manifest'
+  endif
+  libuv_test_exe = executable('uv_run_tests', libuv_test_src,
+    dependencies: [libuv_dep, libuv_test_deps], c_args: libuv_cargs)
+  test('libuv_run_tests', libuv_test_exe, workdir: meson.current_source_dir(),
+    protocol: 'tap', timeout: 600, is_parallel: false)
```

Comment:
Do these need such large timeouts? Probably not.

Seems disappointing if they don't run in parallel. :(

Reply:
Sure, I can lower them.

yeah, I tried to run them in parallel but some tests are binding to ports etc and therefore a couple of EADDRINUSE etc. occur.

Reply:
Why is 300 better than 600? The default is 30, do these need longer? Your response doesn't make it clear if you've timed them and come up with an average reasonable amount of time for them to take...